src/parser/jobs/brd/modules/Snapshots.tsx

```diff
@@ -0,0 +1,260 @@
+	}
+
+	public addSnapshot(snap: CastEvent, playerStatuses: Status[]) {
+		// If IJ didn't refresh a caustic bite, it was a misuse
+		// (IJ on storm bite is technically potency neutral, though odd)
+		const isBadRefresh = (snap.ability.guid === ACTIONS.IRON_JAWS.id)
+							&& !this.statuses.has(STATUSES.CAUSTIC_BITE)
+
+		this.currentSnapshot = {
+			snapEvent: snap,
+			statuses: [...Array.from(this.statuses), ...playerStatuses],
+			badRefresh: isBadRefresh,
+		}
+		this.snapshots.push(this.currentSnapshot)
+	}
+}
+
+export default class Snapshots extends Module {
+	static handle = 'snapshots'
+	static title = t('brd.snapshots.title')`Snapshots`
+	static debug = false
+
+	@dependency private combatants!: Combatants
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+	@dependency private util!: Util
+
+	private targets: Map<string, Target> = new Map()
+
+	protected init() {
+		this.addEventHook('cast', {by: 'player', abilityId: SNAPSHOTTERS}, this.onSnapshot)
+		this.addEventHook(['applydebuff', 'refreshdebuff'], this.onApply)
+		this.addEventHook('removedebuff', this.onRemove)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onApply(event: AbilityEvent) {
+		if (event.targetIsFriendly) { return }
+
+		const target = this.getTarget(event)
+		const status = this.data.getStatus(event.ability.guid)
+		if (status) {
+			target.addStatus(status)
+		}
+	}
+
+	private onRemove(event: AbilityEvent) {
+		if (event.targetIsFriendly) { return }
+
+		const target = this.getTarget(event)
+		const status = this.data.getStatus(event.ability.guid)
+		if (status) {
+			target.removeStatus(status)
+		}
+	}
+
+	private onSnapshot(event: CastEvent) {
+		const target = this.getTarget(event)
+		target.addSnapshot(event, this.playerStatuses)
+	}
+
+	private getTarget(fields: {targetID?: number, targetInstance?: number}): Target {
+		const targetKey = `${fields.targetID}-${fields.targetInstance}`
+
+		if (!this.targets.has(targetKey)) {
+			const targetName = this.parser.pull.actors
+				.find(actor => actor.id === fields.targetID?.toString())
+				?.name ?? 'Unknown Enemy' // Default to "Unknown Enemy" if we can't find a name
+
+			this.targets.set(targetKey, new Target(targetName, targetKey))
+		}
+
+		return this.targets.get(targetKey)!
+	}
+
+	private get playerStatuses(): Status[] {
+		const statuses = this.combatants.selected.getStatuses()
+			.map((event: AbilityEvent) => this.data.getStatus(event.ability.guid))
+			.filter((status: Status | null) => status != null && !SNAPSHOT_BLACKLIST.includes(status.id))
+
+		return statuses
+	}
+
+	private onComplete() {
+		const badRefreshes = Array.from(this.targets.values(), target => target.snapshots)
+			.reduce((acc, snap) => acc.concat(snap))
+			.filter(snap => snap.badRefresh)
+			.length
+
+		const moduleLink = <a href="javascript:void(0);" onClick={() => this.parser.scrollTo(Snapshots.handle)}><NormalisedMessage message={Snapshots.title}/></a>
```

Comment:
React is complaining:
> Warning: A future version of React will block javascript: URLs as a security precaution. Use event handlers instead if you can. If you need to generate unsafe HTML try using dangerouslySetInnerHTML instead. React was passed "javascript:void(0);".

I know we use this elsewhere in the codebase, but we really need to move away from doing it, so we should avoid it here.  Using `a style={{cursor: 'pointer'}}` (or better yet, a custom component that handles the onClick and the pointer styling itself and can be reused in other places) keeps the "this is a link" pointer without using the JS URL that's going to break on us in the future.